### PR TITLE
Make archive sub attrib optional in packages sources attributes in package YAML file

### PR DESCRIPTION
**Pull request by alanoe**

### CODE REVIEW

File: lib/repository.py [hunk 1]
```diff
@@ -41,6 +42,13 @@ def get_git_repository(name, remote_repo_url, parent_dir_path):
     dir, according to specified name.
     If it does not exist, clone it from the specified URL.
     """
+    # infer name if it is not provided
+    if not name:
+        match_obj = re.match(".*/([a-zA-Z0-9\-]*)\.git", remote_repo_url)
```
Comment (ghost) [inline lib/repository.py]: Won't this break if the repository name is some weird string?

There are some [special characters that are allowed in a URL](http://stackoverflow.com/questions/1856785/characters-allowed-in-a-url#1856809) that could show up in the repository's name.

**Review by ghost — CHANGES_REQUESTED** [review]
I'd rather not pass in 'None' into all those function calls, just to have an 'if' checking if the name was provided.

Better to just remove the 'name' parameter and either infer it or fail noisily.

File: lib/repository.py (hunk 1 above)
Comment (ghost) [reply lib/repository.py]: I'd use `urlparse` combined with `os.path.basename` for that.

**Comment by olavphibm** [comment]
start build

**Comment by ghost** [comment]
I'm with either removing the "name" parameter completely, or leaving it as a keyword arg, so we don't have to pass `None` every time.

**Comment by olavphibm** [comment]
start build

File: lib/repository.py (hunk 1 above)
Comment (alanoe) [reply lib/repository.py]: imho [a-zA-Z0-9\-_]* is a safe regex, I have never seen other chars in source code repositories names

**Comment by alanoe** [comment]
OH> I'm with either removing the "name" parameter completely, or leaving it as a keyword arg, so we don't have to pass None every time.

Agreed. I see no reason to have this argument in our build code, so I'll remove it.

File: lib/repository.py [hunk 2]
```diff
@@ -35,12 +36,15 @@ def __init__(self, push_info):
...
+def get_git_repository(remote_repo_url, parent_dir_path):
     """
-    Get a local git repository located in a subdirectory of the parent
-    dir, according to specified name.
+    Get a local git repository located in a subdirectory of the parent directory.
     If it does not exist, clone it from the specified URL.
     """
+    # infer name
+    url_parts = urlparse.urlparse(remote_repo_url)
+    name = os.path.basename(url_parts.path)
```
Comment (ghost) [inline lib/repository.py]: This name might have a ".git" that we need to remove.

File: lib/repository.py [hunk 3]
```diff
@@ -35,12 +36,15 @@ def __init__(self, push_info):
         super(PushError, self).__init__(message)
 
 
-def get_git_repository(name, remote_repo_url, parent_dir_path):
+def get_git_repository(remote_repo_url, parent_dir_path):
     """
-    Get a local git repository located in a subdirectory of the parent
-    dir, according to specified name.
+    Get a local git repository located in a subdirectory of the parent directory.
```
Comment (ghost) [inline lib/repository.py]: You could mention it is "named after the base name of the URL path"

File: lib/versions_repository.py [hunk 4]
```diff
@@ -15,12 +15,12 @@ def setup_versions_repository(config):
     Clone and checkout the versions repository and halt execution if
     anything fails.
     """
-    path, dir_name = os.path.split(
+    path, _ = os.path.split(
```
Comment (ghost) [inline lib/versions_repository.py:18]: `os.path.dirname` could be used here instead of `split`

**Review by ghost — CHANGES_REQUESTED** [review]

File: lib/versions_repository.py (hunk 4 above)
Comment (alanoe) [reply lib/versions_repository.py:18]: I prefer to not alter how parent directory path is obtained in this patch. This is an improvement which could be done in a separate patch.

File: lib/repository.py (hunk 2 above)
Comment (alanoe) [reply lib/repository.py]: Done

File: lib/repository.py (hunk 3 above)
Comment (alanoe) [reply lib/repository.py]: Done

File: lib/repository.py (hunk 2 above)
Comment (ghost) [reply lib/repository.py]: I think that `# infer name` could be removed.

It doesn't add much information to that context. It ends up being just noise in the source code.

**Comment by fabianorosas** [comment]
start build

**Comment by alanoe** [comment]
start build

File: lib/repository.py [hunk 5]
```diff
@@ -35,12 +36,16 @@ def __init__(self, push_info):
...
-def get_git_repository(name, remote_repo_url, parent_dir_path):
+def get_git_repository(remote_repo_url, parent_dir_path):
     """
-    Get a local git repository located in a subdirectory of the parent
-    dir, according to specified name.
+    Get a local git repository located in a subdirectory of the parent directory,
+    named after the file name of the URL path (git default).
     If it does not exist, clone it from the specified URL.
     """
+    # infer name
```
Comment (ghost) [inline lib/repository.py]: Please remove this comment, it doesn't add meaningful information.

**Comment by ghost** [comment]
start build

**Review by ghost — APPROVED** [review]

**Comment by ghost** [comment]
start build

**Comment by farosas** [comment]
This relates to #38

File: lib/repository.py (hunk 5 above)
Comment (alanoe) [reply lib/repository.py]: IMHO considering a comment to be obvious (and thus useless) or not is usually a matter of personal taste. Very minor issue IMHO. Anyway, I have updated the comment to make it more useful.

**Comment by alanoe** [comment]
Pls let us merge this ASAP so I do not need to rebase this patch again.

**Review by ghost — APPROVED** [review]

**Pull request merged**